16.0 mig account invoice merge auto

fcayre opened the pull request.

mourad-ehm committed:
[FIX] move to new folder account_invoice_merge_auto

mourad-ehm committed:
[ADD] account_invoice_merge_auto: Migration to 12.0

mourad-ehm committed:
[FIX] move _cron_invoice_merge & _invoice_merge_candidates from res.partner to account.invoice

mourad-ehm committed:
[FIX] pep8

fcayre committed:
[FIX] Broken test and incorrect cron definition

Finish the work started at e26b706f166717a63f560ce87a7a500507b2e110:
- remove duplicated code from res partner
- fix test accordingly
- fix cron definition to v12 api

fcayre committed:
[FIX] Review readme

Contributors were wrong and pasted references to field-service had to be fixed.

fcayre committed:
[IMP] Use oca-gen-addon-readme to generate a clean README.rst

fcayre committed:
[FIX] Move the auto merge related fields to an always visible of the partner form

fcayre committed:
Remove partner's auto_merge_invoice field

It was supposed to be a web UI only field, but:
1. a typo made the UI feature partly working (typo in the invisible view field attributes)
2. this field was ignored in the invoice merge mecanism itself.

We now use invoice_merge_next_date for both the UI and the feature implementation, which
is simpler and less confusing for the users.

fcayre committed:
Fix manifest's website

fcayre committed:
[IMP] account_invoice_merge_auto: black, isort, prettier

fcayre committed:
[MIG] account_invoice_merge_auto: Migration to 16.0

fcayre committed:
Impact invoice model an view changes

fcayre committed:
Fix tests when installed with a non-default account chart template

fcayre committed:
[tests] Add tests for partner invoice_merge_time_interval

9184Lucky commented on account_invoice_merge_auto/tests/test_res_partner.py:
```diff
@@ -0,0 +1,15 @@
+        self.assertEqual(_date + comp_interval("daily", 10), date(2025, 1, 11))
+        self.assertEqual(_date + comp_interval("weekly", 2), date(2025, 1, 15))
+        self.assertEqual(_date + comp_interval("monthly", 2), date(2025, 3, 1))
+        self.assertEqual(_date + comp_interval("monthlylastday", 2), date(2025, 3, 31))
```
à mon avis ça doit pas être trop dure d'aller gratter le "else" là....

fcayre replied:
Tout à fait je l'avais même déjà fait et pas encore poussé :smile:

fcayre committed:
Refactor condition as a boolean expression

This is simpler and OK here (a real date cannot be falsy).
It improves readability and avoids a pragma: no cover to exclude
the line from code coverage.

Honeyxilia commented on account_invoice_merge_auto/models/res_partner.py:
```diff
@@ -33,7 +33,7 @@ class InvoiceMergeAutoPartner(models.Model):
 
     invoice_merge_next_date = fields.Date(
-        String="Invoice merge next date",
+        string="Invoice merge next date",
```
Je pense que cette modification serait plus pertinante dans le commit de linting initial

fcayre committed:
[tests] Rewrite test using helper class from the account module

fcayre commented:
/ocabot merge patch
